feat: add third release blog post (#16)
ariel-os/ariel-os.org: content/blog/ariel-os-0.3.0.md

# tags = ["Release"]
+++

Today we are happy to announce a new version of Ariel OS. With over 6 months of work and 400 merged PRs, this new version brings shiny features and improvements. These include upgraded Embassy and esp-hal dependencies, UART support, a brand new way to write and access sensors, Bluetooth Low Energy support, structured board descriptions and much more!

On top of those new features, this release adds support to more MCUs and boards, some of those were added by the community!

You can check out the full changelog of [Ariel OS 0.3.0](https://github.com/ariel-os/ariel-os/releases/tag/v0.3.0).

Let's go over some of the highlights of this release.

## Upgraded Embassy and esp-hal

This was the major blocker for this new release, making everything work with as little breakage as possible, most applications built for Aiel OS v0.2.0 should work with Ariel OS v0.3.0 with little to no modification. You can check the full list of breaking changes in [the changelog](https://github.com/ariel-os/ariel-os/releases/tag/v0.3.0).

Ariel OS v0.3.0 uses `esp-hal` v1.0.0, `embassy-nrf` v0.8.0, `embassy-rp` v0.8.0, `embassy-stm32` v0.4.0.

> With esp-hal becoming stable, upgrades to newer versions should come faster. Embassy isn't stable yet but now has a faster release schedule that should make upgrades faster and easier for us.

Reviewer: Maybe

```suggestion
With esp-hal having stabilized as 1.0, upgrades to newer versions should come faster. Embassy has not reached that milestoneyet, but now has a faster release schedule that should make upgrades faster and easier for us.
```

This phrasing might help not making it about stability (which is a somewhat ambiguous term out of the context of semver) and more about versioning.

## UART support

UART peripherals are now handled directly by Ariel OS when supported, you can check if your board is supported [in the hardware support chapter of our book](https://ariel-os.github.io/ariel-os/dev/docs/book/hardware-functionality-support.html). This allows you to easily write and read from UART, without having to worry about which hardware and HAL will be used.

Basic usage looks like this (full example in [`tests/uart-loopback`](https://github.com/ariel-os/ariel-os/tree/main/tests/uart-loopback)):

> ```rust

Author: Will be handled by adding more padding with CSS.

+++
date = '2026-01-27T15:28:40+01:00'
draft = false
title = "Ariel OS v0.3.0: BLE, Sensors, UART, and More!"
# tags = ["Release"]
+++

Today we are happy to announce a new version of Ariel OS!
With over 6 months of work and 400 merged PRs, this new version brings shiny new features and improvements.
These include upgraded Embassy and esp-hal dependencies, UART support, a brand new way to write and access sensor drivers, Bluetooth Low Energy support, structured board descriptions, and much more!

On top of those new features, this release adds support for more microcontrollers and boards, and some of those were added by the community!

You can check out the [full changelog of Ariel OS v0.3.0](https://github.com/ariel-os/ariel-os/blob/v0.3.0/CHANGELOG.md#030---2026-02-02).

Let's go over some of the highlights of this release.

## Upgraded Embassy and esp-hal

> This was the major blocker for this new release, making everything work with as little breakage as possible, most applications built for Ariel OS v0.2.1 should work with Ariel OS v0.3.0 with little to no modifications. You can check the full list of breaking changes in [the changelog](https://github.com/ariel-os/ariel-os/blob/v0.3.0/CHANGELOG.md#030---2026-02-02).

Reviewer 2: ```suggestion
This was the major blocker for this new release, making everything work with as little breakage as possible: most applications built for Ariel OS v0.2.1 should work with Ariel OS v0.3.0 with little to no modifications. You can check the full list of breaking changes in [the changelog](https://github.com/ariel-os/ariel-os/blob/v0.3.0/CHANGELOG.md#030---2026-02-02).
```

> Ariel OS v0.3.0 uses `esp-hal` v1.0.0, `embassy-nrf` v0.8.0, `embassy-rp` v0.8.0, `embassy-stm32` v0.4.0.

Reviewer: ```suggestion
Ariel OS v0.3.0 publicly uses `esp-hal` v1.0.0, `embassy-nrf` v0.8.0, `embassy-rp` v0.8.0, `embassy-stm32` v0.4.0.
```

For experienced Rustaceans who enter this project, this may be an important piece of information (if it is accurate; I think it is even though I'd prefer we change it on the long run).
